be/src/olap/storage_engine.cpp

```diff
@@ -862,11 +862,13 @@ void StorageEngine::_parse_default_rowset_type() {
 }
 
 void StorageEngine::start_delete_unused_rowset() {
-    std::lock_guard<std::mutex> lock(_gc_mutex);
-    for (auto it = _unused_rowsets.begin(); it != _unused_rowsets.end();) {
-        if (it->second.use_count() != 1) {
-            ++it;
-        } else if (it->second->need_delete_file()) {
+    std::unordered_map<std::string, RowsetSharedPtr> unused_rowsets_copy;
+    {
+        std::lock_guard<std::mutex> lock(_gc_mutex);
+        unused_rowsets_copy = _unused_rowsets;
+    }
+    for (auto it = unused_rowsets_copy.begin(); it != unused_rowsets_copy.end(); ++it) {
+        if (it->second.use_count() == 1 && it->second->need_delete_file()) {
```

Comment:
I am not sure that use_count is at least 2 after 'unused_rowsets_copy = _unused_rowsets'.